docs/YARN.md

```diff
@@ -0,0 +1,294 @@
+> to get confusing if people start trying to use other package managers. It’s
+> much simpler just to put it all in the same script definition, or outsource to
+> a script file.
+
+## Check it works
+
+At this point you should have a working setup.
+
+## Remove shx or equivalents (probably)
+
+According to
+[this blog post](https://dev.to/arcanis/introducing-yarn-2-4eh1#normalized-shell):
+
+> Yarn 2 ships with a rudimentary shell interpreter that knows just enough to
+> give you 90% of the language structures typically used in the scripts field.
+> Thanks to this interpreter, your scripts will run just the same regardless of
+> whether they're executed on OSX or Windows.
+
+So you probably don’t need `shx` or any other tool that you were using for
+cross-platform shell compatibility. Then again, maybe you do still need it
+because you use a language structure outside the 90%. 🤷
+
+## Consider using constraints
+
+E.g. to ensure that each package specifies the right fields in its
+`package.json`. Bonus: you get to learn Prolog!
+
+https://yarnpkg.com/features/constraints
+
+## TypeScript plugin
+
+Automatically installs `@types` definitions if the project doesn’t have its own.
+
+```sh
+yarn plugin import typescript
+```
+
+https://github.com/yarnpkg/berry/tree/master/packages/plugin-typescript
+
+## Version plugin
+
+```sh
+yarn plugin import version
+```
+
+## interactive-tools plugin
+
+Most useful for `yarn upgrade-interactive`
+https://yarnpkg.com/cli/upgrade-interactive
+
+## Plug-n-play and node_modules
+
+Read about the problems of `node_modules` and how Yarn aims to fix them here:
+https://yarnpkg.com/features/pnp#the-node_modules-problem
+
+### Check whether your project is ready
+
+```sh
+yarn dlx @yarnpkg/doctor@2
+```
+
+> NOTE: You need to specify the `@2` because they released the incompatible v3
+> RC as `latest` on npm.
+
+By the way, this uses another new feature of Yarn v2: `yarn dlx`. This is like
+`npx`, but ...
+
+### Remove the `nodeLinker` setting from `.yarnrc.yml`.
+
+Uses default PnP setting.
+
+### Install again
+
+Run `yarn`. This will remove the `node_modules/` directory.
+
+### Update the VCS ignore list
+
+Remove what you put there before and add these:
+
+```
+.yarn/*
+!.yarn/cache
+!.yarn/patches
+!.yarn/plugins
+!.yarn/releases
+!.yarn/sdks
+!.yarn/versions
+```
+
+Basically we’re adding the cache and the `.pnp.js` file to VCS.
+
+### Run node using yarn node
+
+Replace calls to node outside of a `package.json` script with `yarn node`. For
+example in shebangs in development scripts. (You probably don’t want to update
+executables you will ship to your users because they may not be using Yarn.)
+
+## Switch versioning from Lerna to Yarn
+
+The Lerna team seems to be pretty ambivalent about its future. After a period of
+hibernation, they released v4 on 10th February 2021. But the main `README.md`
+for the project still refers to v3, and
```

Comment:
Could we formulate this section more a bit more neutral? I was reading stories about one or two people working on it with one person long gone and the other completely burned out. Let's try to avoid any unnecessary pressure on those people. I think it is safe to simply call it unmaintained after security advisories have not been fixed for months and there is no commit since that one spike in February.